Docs: Renovate

Lunarnovaa opened the pull request:
General PR to address some current issues of the docs. Namely:

- [x] Improve general cohesion of documentation
- [x] Add documentation to our docs page
- [x] Fix the custom link (CNAME)
- [x] Syntax highlighting
- [ ] Clean up the docs more generally, unify tone and such
- [x] Break up CONTRIBUTING.md into multiple smaller files that link together (and modify the sidebar to make this cleaner)
- [x] Add a simple index.md landing page

Opened new PR to allow for me to host the current version at [rum.aurabora.org](https://rum.aurabora.org/index.html).

You can also see the modified readme/GitHub page at [lunarnovaa/hjem-rum](https://github.com/Lunarnovaa/hjem-rum/tree/renovate-docs-upstream).

### Current Blockers:

- [x] New SNUG website domain
- [x] NDG additive stylesheet option

### Meta
Related Issue: #73

[We do not currently have a policy against AI-generated code, but we ask you to disclose the usage of AI for code generation]::
AI used to generate code included in this PR?: No

### All Submissions:
 - [x] Formatted commit message in accordance with CONTRIBUTING.md guidelines
 - [x] Filled in all meta items
 - [x] Mentioned any blockers before the PR can merge
 - [x] Verified there are no conflicting PRs open

Lunarnovaa committed:
docs: update CNAME for upstream adoption

llakala commented:
Is it possible to sort these better?
![Screenshot_2025-06-23-16-28-09-43_3aea4af51f236e4932235fdada7d1643](https://github.com/user-attachments/assets/0cc02b7c-80a3-48aa-b184-ff539aee3d0c)
Subsections would also be nice (Contributing vs Usage). Not sure if that's possible.

Lunarnovaa committed:
meta: init docs testing workflow

Lunarnovaa committed:
modules: fix descriptions

desktops/hyprland: clarify file

programs/fish: prevent broken auto-linking

Lunarnovaa committed:
docs: init CNAME

Lunarnovaa committed:
docs: init footer

Lunarnovaa committed:
docs/package: cleanup

Lunarnovaa committed:
docs: minor improvements

docs: change option depth from 2->3

docs: init highlighting code

previously, code highlighting was not working upstreamed, but it is now
being worked on. this pr will not be merged until it is working

Lunarnovaa committed:
docs: fix footer formatting

Lunarnovaa committed:
docs: major renovation of all documenation

docs/README: move to .github/

docs/README: pare down to basic introduction

docs/README: init pronunciation info/helper

docs/README: init FAQ

docs/CONTRIBUTING: break into multiple files

docs/CONTRIBUTING: init DOCS

docs/USAGE: move usage README section into INSTALLATION and QUIRKS

docs: init informative pages on published site

docs: general rewriting, additions, and improvements

docs: add anchor links

docs/writing-modules: add info on nullable packages

docs/writing-modules: add dependence on config section

docs/installation: add info on configuring rum in hjem modules

docs/installation: add info on smfh

Lunarnovaa committed:
docs: update ndg

Lunarnovaa committed:
docs: init cc-by-nc-sa license

docs/README: modify GPLv3 explanation + link(s)

docs/footer: add license info

Lunarnovaa committed:
docs/README: change links to fork

Lunarnovaa committed:
docs/README: fix éclairevoyant name

a-usr commented:
I got a notification that someone commented on a commit, but that commit is gone, so I cant see it. What did they say?
